### ISSUE TITLE
relocating type "R_AARCH64_CALL26" not found!!!

### ISSUE THREAD

**Issue by saliey** [upx/upx#184]
When I was doing the processing of the arm64 elf file, there was a "R_AARCH64_CALL26" relocating type not found. Is there any solution?

**Comment by jreiser** [comment]
Support for R_AARCH64_CALL26 was added to the <tt>devel</tt> branch (file src/linker.cpp) on 2017-05-06, ten months ago.  The https://github.com/upx/upx-stubtools do not generate that relocation style when processing the source to upx, so some other toolchain (compiler, assembler, etc.) must have been used to build the upx that generated the message.

**Comment by m-flak** [comment]
Right or Wrong?: Its an arm64 ABI specific relocation, or whatever. So, arm64 build of the genn'ing upx will be able to process no problemo. Cross Compile to Google's toolchain if you're working with an Android binary— whatever's native, you get the picture. TBH I never seen this kinda reloc, but I enjoy x86_64/amd64 ABI.

**Comment by jreiser** [comment]
Yes, R_AARCH64_CALL26 is specific to aarch64.  Independent of host, each upx can process for all target architectures.  Get the <tt>devel</tt> branch.

**Comment by saliey** [comment]
I ran into one other problem, and I added my own code to the arm.v4t-linux.shlib-init.s file, which was compiled with the shell elf file, but crashed when I was running.I use IDA debugging, found in ` ` ` LDMIA R1!, {R2-R4,R6-R9,R12}
SUBS R5, R5, #1
STMIA R0!, {R2-R4,R6-R9,R12}
BNE loc_EEC574BC
LDMFD SP!, {R2,R4,R6-R9}` ` ` code in duplicate, the original instructions' D4 03 BD E8 'has been changed to' F0 01 F0 E7 ', what reason is this

At 2018-03-10 12:58:38, "Matthew E. Kehrer" <notifications@github.com> wrote:


Right or Wrong?: Its an arm64 ABI specific relocation, or whatever. So, arm64 build of the genn'ing upx will be able to process no problemo. Cross Compile to Google's toolchain if you're working with an Android binary— whatever's native, you get the picture. TBH I never seen this kinda reloc, but I enjoy x86_64/amd64 ABI.

**Comment by saliey** [comment]
I ran into one other problem, and I added my own code to the arm.v4t-linux.shlib-init.s file, which was compiled with the shell elf file, but crashed when I was running.I use IDA debugging, found in ` ` ` LDMIA R1!, {R2-R4,R6-R9,R12}
SUBS R5, R5, #1
STMIA R0!, {R2-R4,R6-R9,R12}
BNE loc_EEC574BC
LDMFD SP!, {R2,R4,R6-R9}` ` ` code in duplicate, the original instructions' D4 03 BD E8 'has been changed to' F0 01 F0 E7 ', what reason is this

在 2018-03-10 21:03:30，"John Reiser" <notifications@github.com> 写道：


Closed #184.

**Comment by m-flak** [comment]
Think u have to just like guess that one out.

Magic number gaussing is an art, like all human behavior- thankfully easy
to predict That's the blanket over the mirror here: It's a value , sourced
from your mind seconds thot, irrationally acquired.

*tldr F0F01F0E7 resembles 'FOREVER1' & 'FOREVERONE7'*
*F and 7 hexs are the degaussed, non-imaginary originii for that DWORD bro.
Just look at it, it's a hexadecimaltatic flying crarpet.*
